### PR TITLE
Update cyclone tracker to handle return code correctly

### LINKED ISSUE #507: Verify should exit non-zero if files are never found
The verify job currently does not exit with a non-zero error code if files are missing. data_check_gfs.sh correctly exits with a non-zero code, but the calling script does not pass that along.

```
+ data_check_gfs.sh line 50: msg='FATAL ERROR: 9 files missing in /work/noaa/marine/jmeixner/workflowPR500/test01/COMROOT/test01/gfs.20130101/00/atmos'
 + data_check_gfs.sh line 51: echo 'FATAL ERROR: 9 files missing in /work/noaa/marine/jmeixner/workflowPR500/test01/COMROOT/test01/gfs.20130101/00/atmos'
FATAL ERROR: 9 files missing in /work/noaa/marine/jmeixner/workflowPR500/test01/COMROOT/test01/gfs.20130101/00/atmos
 + data_check_gfs.sh line 51: postmsg /work/noaa/marine/jmeixner/workflowPR500/test01/EXPROOT/test01/logs/jlogfile 'FATAL ERROR: 9 files missing in /work/noaa/marine/jmeixner/workflowPR500/test01/COMROOT/test01/gfs.20130101/00/atmos'
 + data_check_gfs.sh line 52: exit 6
3602.508 + [ 6 -eq 6 ]
3602.508 + exit
3602.805 + err=0
3602.805 + export err
3602.805 + err_chk
3602.811 + [ -e OUTPUT.203671 ]
3602.811 + msg='ENDED NORMALLY.'
3602.811 + postmsg /work/noaa/marine/jmeixner/workflowPR500/test01/EXPROOT/test01/logs/jlogfile 'ENDED NORMALLY.'
3602.819 + cd /work/noaa/stmp/jmeixner/RUNDIRS/test01/2013010100/gfs
3602.819 + [[ NO == NO ]]
3602.819 + rm -rf /work/noaa/stmp/jmeixner/RUNDIRS/test01/2013010100/gfs/gfs_cyclone_genesis214996
3602.823 + date
Tue Nov 16 17:22:21 CST 2021
3602.826 + exit 0
```

### PR DESCRIPTION
**Pull request by lgannoaa**
Implement exception handling in JGFS_ATMOS_CYCLONE_TRACKER to handle error code return from ${USHens_tracker}/data_check_gfs.sh

Fix: #507

**Description**
${USHens_tracker}/data_check_gfs.sh return error code if required file not found.
However, jjob JGFS_ATMOS_CYCLONE_TRACKER does not handle this exception and continue process with this silent fail.
Exception handling code is now included.

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: jobs/JGFS_ATMOS_CYCLONE_TRACKER [hunk 1]
```diff
@@ -128,8 +128,7 @@ export loopnum=1
 
 #-----------input data checking -----------------
 ${USHens_tracker}/data_check_gfs.sh
-# exit code 6 = missing data of opportunity
-if [ $? -eq 6 ]; then exit; fi
+export err=$?; err_chk
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGFS_ATMOS_CYCLONE_TRACKER:131]: Should code 6 still be ignored? If the comment is correct, I don't know that we want to fail the job if data of opportunity is missing but everything else is correct.

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
Might need more information about the error codes (particularly 6) from data_check_gfs.sh to make this decision.

File: jobs/JGFS_ATMOS_CYCLONE_TRACKER (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGFS_ATMOS_CYCLONE_TRACKER:131]: Can I get an answer to this so we can get this PR in

Comment (lgannoaa) [reply jobs/JGFS_ATMOS_CYCLONE_TRACKER:131]: Code 6 here is not consider as failure. It can be ignored to make exception handling effective.

Comment (WalterKolczynski-NOAA) [reply jobs/JGFS_ATMOS_CYCLONE_TRACKER:131]: Okay, this needs to be updated then to retain that from the previous code.

**Comment by aerorahul** [comment]
Any progress on this PR?

**Comment by lgannoaa** [comment]
I provided my solution. Do you still have any question @WalterKolczynski-NOAA ?

**Comment by aerorahul** [comment]
@lgannoaa 
The original script exits only if error code is 6.
The change implies all error codes (except 0) will cause an exit.  I think that is reasonable.

I also note that https://github.com/NOAA-EMC/global-workflow/blob/fc6e1c3316397432d2da218b3c4fc0ea1fe70644/jobs/JGFS_ATMOS_CYCLONE_TRACKER#L131
does the check this script was performing.  Would an `exit 6` be appropriate for the above line instead of just `exit`?

**Comment by lgannoaa** [comment]
@aerorahul, exit code 6 cause job to fail. Therefore, job should fail. Hence my PR to modify it.

File: jobs/JGFS_ATMOS_CYCLONE_TRACKER (hunk 1 above)
Comment (aerorahul) [reply jobs/JGFS_ATMOS_CYCLONE_TRACKER:131]: I think this works.

**Review by aerorahul — APPROVED** [review]
looks good

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**